tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+
+from adhocracy4.comments.models import Comment
+from tests.apps.questions.models import Question
+
+
+@pytest.mark.django_db
+def test_serializer(apiclient, question, comment_factory, user):
+    question_ct = ContentType.objects.get_for_model(Question)
+    comment_ct = ContentType.objects.get_for_model(Comment)
+    moderator = question.project.moderators.first()
+
+    comment_factory(pk=1, content_object=question, creator=user)
+    comment_factory(pk=2, content_object=question, creator=user,
+                    comment_categories='QUE')
+    comment_factory(pk=3, content_object=question, creator=user,
+                    is_blocked=True)
+    comment_factory(pk=4, content_object=question, creator=user,
+                    is_censored=True)
+    comment_factory(pk=5, content_object=question, creator=user,
+                    is_removed=True)
+    comment_factory(pk=6, content_object=question, creator=user,
+                    is_moderator_marked=True)
+    comment = comment_factory(pk=7, content_object=question, creator=moderator)
+    comment_factory(pk=8, content_object=comment, creator=user)
+
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    response = apiclient.get(url)
+
+    comment_data = response.data['results']
+    assert len(comment_data) == 7
+
+    comment_1 = comment_data[6]
+    assert comment_1['id'] == 1
+    assert comment_1['user_name'] == ''  # should be user.username ?!!
```

Comment:
Regarding the question: Is this part of react, i.e. comment_list.jsx tested in test_templatetags.py and thus not mocked here?

Reply:
These are my thoughts on the changes I implemented:

- `is_removed` and `is_censored` are exactly the same in the context of the other serializer fields, except for those two fields. The difference is who can perform the action, which is already tested in **test_api.py**.
- `last_discussed` is tested in **test_signals.py**
- I added one comment for testing the comment categories because they are set in **settings.py** and might be needed somewhere, but are never tested anywhere thus far.

Thanks for this!! It visualizes how the fields affect each other very well. 
Is there anything else that I missed?

Reply:
Cool, that you tested all the fields!
But I thought you could also find weird things and inconsistencies in the serializer. For example finding out why the user does not have a name here. And what we use is_deleted for, when we also have is_removed and so on. Also, should the user pk and name for blocked comments also change?

Reply:
Maybe put a comment above the assert blocks what it is testing. Like "is_blocked, one category", "child comment", "comment added by moderator" and so on? Then it might be easier to spot the weird things.